x-pack/plugins/observability_solution/investigate_app/public/pages/details/contexts/investigation_context.tsx

```diff
@@ -0,0 +1,230 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License
+ * 2.0; you may not use this file except in compliance with the Elastic License
+ * 2.0.
+ */
+
+import { i18n } from '@kbn/i18n';
+import { type GlobalWidgetParameters } from '@kbn/investigate-plugin/public';
+import { GetInvestigationResponse, InvestigationItem, Item } from '@kbn/investigation-shared';
+import { isEqual } from 'lodash';
+import React, { createContext, useContext, useEffect, useRef, useState } from 'react';
+import { useAddInvestigationItem } from '../../../hooks/use_add_investigation_item';
+import { useAddInvestigationNote } from '../../../hooks/use_add_investigation_note';
+import { useDeleteInvestigationItem } from '../../../hooks/use_delete_investigation_item';
+import { useDeleteInvestigationNote } from '../../../hooks/use_delete_investigation_note';
+import { useFetchInvestigation } from '../../../hooks/use_fetch_investigation';
+import { useKibana } from '../../../hooks/use_kibana';
+import { useUpdateInvestigation } from '../../../hooks/use_update_investigation';
+import { useUpdateInvestigationNote } from '../../../hooks/use_update_investigation_note';
+
+export type RenderedInvestigationItem = InvestigationItem & {
+  loading: boolean;
+  element: React.ReactNode;
+};
+
+interface InvestigationContextProps {
+  investigation?: GetInvestigationResponse;
+  renderableItems: RenderedInvestigationItem[];
+  globalParams: GlobalWidgetParameters;
+  updateInvestigationParams: (params: GlobalWidgetParameters) => Promise<void>;
+  addItem: (item: Item) => Promise<void>;
+  deleteItem: (itemId: string) => Promise<void>;
+  isAddingItem: boolean;
+  isDeletingItem: boolean;
+  // note
```

Comment:
Looks good! As discussed we can later split the context into 2 contexts, `InvestigationItemsContext` and `InvestigationNotesContext`